docker-compose.dev.yml

```diff
@@ -9,9 +9,11 @@
 
 services:
   server:
-    command: npm run build:watch
-    environment:
-      NODE_ENV: development
+    build:
+      dockerfile: Dockerfile.dev
```

Comment:
I think we can get around having to have a separate `Dockerfile.dev` file by using multistage builds and specifying the appropriate `target` in our `docker-compose.*yml` files: https://stackoverflow.com/a/53101932/177677

I think I know what this is trying to accomplish: Make it so that prod only has prod stuff installed, and dev has dev stuff preinstalled. But we're already having some friction with duplicate work taking place in Dockerfiles (see #1640), and I worry that this sets us up for dev and prod environments accidentally diverging from each other if contributors and reviewers fail to catch that an update needs to happen in both files.